### PR TITLE
Seal PropertyGridView

### PR DESCRIPTION
**Pull request by JeremyKuhne**
PropertyGridView had been gaining virtuals for testing and this wasn't necessary. Sealing the class for perf and maintainability.

Rename GridViewEdit to GridViewTextBox to match typical naming. Also change "Edit" to "EditTextBox" to follow.

Other related code cleanup, including better documentation for Control message preprocessing.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5484)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs [hunk 1]
```diff
@@ -3972,16 +3971,16 @@ private void SetupToolbar(bool fullRebuild)
             else
             {
                 // Clear all the items from the toolStrip and image list after the first two.
-                int count = buttonList.Count;
+                int items = buttonList.Count;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:3974]: maybe itemsCount?

**Comment by Tanya-Solyanik** [comment]
Please rename the test file as well Edit->TextBox

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs [hunk 2]
```diff
@@ -794,9 +781,9 @@ private void ClearGridEntryEvents(GridEntryCollection entries, int startIndex, i
             }
         }
 
-        public void ClearProps()
+        public void ClearGridEntries()
         {
-            Debug.WriteLineIf(CompModSwitches.DebugGridView.TraceVerbose, "PropertyGridView:ClearProps");
+            Debug.WriteLineIf(CompModSwitches.DebugGridView.TraceVerbose, "PropertyGridView:ClearClearGridEntries");
```
Comment (Shyam-Gupta) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:787]: typo: ClearClearGridEntries -> ClearGridEntries.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs [hunk 3]
```diff
@@ -904,62 +889,63 @@ private void CommonEditorHide(bool always)
...
+                    gotFocus = User32.SetFocus(new HandleRef(this, Handle)) != IntPtr.Zero;
                 }
             }
 
             try
             {
                 // We do this because the Focus call above doesn't always stick, so
-                // we make the Edit think that it doesn't have focus.  this prevents
-                // ActiveControl code on the containercontrol from moving focus elsewhere
+                // we make the Edit think that it doesn't have focus. This prevents
```
Comment (Shyam-Gupta) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs:905]: nit: Edit -> EditTextBox

**Review by Shyam-Gupta — APPROVED** [review]
LGTM

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:3974]: This got flipped back unexpectedly. Going back to count.

**Comment by JeremyKuhne** [comment]
> Please rename the test file as well Edit->TextBox

@Tanya-Solyanik not clear on what you're referring to here.

**Review by dreddy-work — COMMENTED** [review]
LGTM.

**Review by dreddy-work — DISMISSED** [review]

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewEdit.GridViewEditAccessibleObjectTests.cs [hunk 4]
```diff
@@ -36,7 +36,7 @@ public void GridViewEditAccessibleObject_created_for_string_property()
             UiaCore.IRawElementProviderFragment editFieldAccessibleObject = selectedGridEntryAccessibleObject.FragmentNavigate(UiaCore.NavigateDirection.FirstChild);
             Assert.NotNull(editFieldAccessibleObject);
 
-            Assert.Equal("GridViewEditAccessibleObject", editFieldAccessibleObject.GetType().Name);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewEdit.GridViewEditAccessibleObjectTests.cs:39]: Please rename this file:
`PropertyGridView.GridViewTextBox.GridViewTextBoxAccessibleObjectTests.cs`

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by Shyam-Gupta — APPROVED** [review]

**Pull request merged**